### PR TITLE
176 106 174 149 database orm updates

### LINKED ISSUE #106: DB Squad
The database should be able to handle inventories of items that can be equipped.
As well as how units are grouped in a squads

### LINKED ISSUE #149: Use random names for default new unit names
Pull from the random names table to use of a default name, perhaps a random button when naming a character

### LINKED ISSUE #174: Connect Squads to Database
- Add squads to database on creation
- Add name and description to squads

### LINKED ISSUE #176: Town Data Storing DB/ORM
-Location
-Name
-Units inside

### PR DESCRIPTION
**Pull request by Toeecko**
I know, kinda crazy 4 issues in one branch and pull request. But they are all so related and build off of each other. So I would have to pull each one in the right order into each other. They are also relatively small.

Fixes #176 
Fixes #106
Fixes #174
Fixes #149

Changes proposed in this pull request:
- Adds Squad Table and ORM
- Adds Town Table and ORM
- Adds Random Name Generation

### CODE REVIEW

File: Assets/Scripts/Battlefield/CreatureScripts/Ability.cs [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
+    public class Ability : AbstractAbility
     {
+        /*
+        public Ability(CombatAbilities abilityStats)
+        {
+
+        }*/
+
+
+
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/CreatureScripts/Ability.cs:16]: Can this be cleaned up?

File: Assets/Scripts/Battlefield/Enemies/EnemyFactory.cs [hunk 2]
```diff
@@ -51,16 +51,16 @@ void Awake()
 
                 //UniqueCreature
                 uniqueCreature.creatureName = enemyData.Name;
-                uniqueCreature.maxHealth = enemyData.Stats.HP;
+                uniqueCreature.maxHealth = enemyData.Stats.Max_HP;
                 uniqueCreature.maxMovement = enemyData.Stats.Movement;
                 uniqueCreature.isEnemy = true;
-
+
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/Enemies/EnemyFactory.cs:57]: Bad spacing

File: Assets/Scripts/Battlefield/CreatureScripts/CharacterFactory.cs [hunk 3]
```diff
@@ -66,6 +66,11 @@ void Awake()
                 stats.evasion = unitTable.Stats.Evasion;
                 stats.role = unitTable.Role.Name;
 
+                //Abilities
+
+                abilities.abilities.Add(new Ability());
+
+
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/CreatureScripts/CharacterFactory.cs:73]: Bad spacing

File: Assets/Scripts/Battlefield/CreatureScripts/AbstractAbility.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace SwordAndBored.Battlefield.CreaturScripts { 
+    public abstract class AbstractAbility
+    {
+        public string AttackName;
+
+        public abstract void Initialize(GameObject obj);
+        public abstract void TriggerAbility(GameObject target);
+        
+
```
Comment (wynn-rj) [inline Assets/Scripts/Battlefield/CreatureScripts/AbstractAbility.cs:13]: Bad spacing

**Review by cthomas68 — APPROVED** [review]

File: Assets/Scripts/Battlefield/CreatureScripts/Ability.cs (hunk 1 above)
Comment (Toeecko) [reply Assets/Scripts/Battlefield/CreatureScripts/Ability.cs:16]: That's from battlefield, not part just part of earlier branch. Is resolved in alternate branch

File: Assets/Scripts/Battlefield/Enemies/EnemyFactory.cs (hunk 2 above)
Comment (Toeecko) [reply Assets/Scripts/Battlefield/Enemies/EnemyFactory.cs:57]: Same as above

File: Assets/Scripts/Battlefield/CreatureScripts/CharacterFactory.cs (hunk 3 above)
Comment (Toeecko) [reply Assets/Scripts/Battlefield/CreatureScripts/CharacterFactory.cs:73]: Same as above

File: Assets/Scripts/Battlefield/CreatureScripts/AbstractAbility.cs (hunk 4 above)
Comment (Toeecko) [reply Assets/Scripts/Battlefield/CreatureScripts/AbstractAbility.cs:13]: Same as above

**Review by wynn-rj — APPROVED** [review]

**Pull request merged**